### ISSUE TITLE
ExperimentalKVStore: Move withRead/withWrite to Replicache

### ISSUE THREAD

**Issue by aboodman** [rocicorp/replicache#477]
Why do clients have to implement this? It seems like given read(), write(), and release(), we can do this ourselves.

**Comment by arv** [comment]
Yup. I was hoping we could only have, `withRead` and `withWrite` (and no `read`, `write` and `release`). I think it is still doable if I refactor transactions to remove some intermediate abstractions.
